src/v2/providers/pubsub.ts

```diff
@@ -164,5 +166,25 @@ export function onMessagePublished<T = any>(
     },
   });
 
+  const baseOpts = options.optionsToEndpoint(options.getGlobalOptions());
+  const specificOpts = options.optionsToEndpoint(opts);
+
+  const endpoint: ManifestEndpoint = {
+    platform: 'gcfv2',
+    ...baseOpts,
+    ...specificOpts,
+    labels: {
+      ...baseOpts?.labels,
+      ...specificOpts?.labels,
+    },
+    eventTrigger: {
+      eventType: 'google.cloud.pubsub.topic.v1.messagePublished',
+      eventFilters: { topic },
+      retry: false,
+    },
+  };
+  copyIfPresent(endpoint.eventTrigger, opts, 'retry', 'retry');
+  func.__endpoint = endpoint;
```

Comment:
We seem to be alternating between using `Object.defineProperty()` and directly setting the property in the v2 providers. Would it be beneficial to stick with one format for all the providers?

Reply:
I have mixed feels. We could always use define property because it's safest, but it is a bit arcane and is only needed for endpoint types that rely on firebaseConfig (RTDB and Storage)

Reply:
I've been leaning towards "don't use `defineProperty()` unless we have to. And we have to use `defineProperty()` in some cases b/c user code will throw an error if environment variables like `FIREBASE_CONFIG` does not exist (which we use to define a trigger, like default bucket info).